cdap-watchdog/src/main/java/co/cask/cdap/logging/logbuffer/LogBufferWriter.java

```diff
@@ -169,7 +180,10 @@ private Location rotateFile(OutputStream currOutputStream) throws IOException {
     currOffset = 0;
     // update current file id to next monotonically increasing file id
     currFileId = currFileId + 1;
-    return locationFactory.create(getFileName(currFileId));
+    Location rotatedLocation = locationFactory.create(getFileName(currFileId));
+    // executes log buffer cleaner runnable
+    executorService.execute(cleaner);
```

Comment:
Executing the cleaner on every rotation may not be a good idea when there is a heavy load on writes and hence fast rotations.

Reply:
Hmm, good point. Fixed